llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -49253,8 +49253,10 @@ static SDValue combineX86SubCmpForFlags(SDNode *N, SDValue Flag,
     return SDValue();
 
   SDValue X = SetCC.getOperand(1);
-  // Replace API is called manually here b/c the number of results may change.
-  DAG.ReplaceAllUsesOfValueWith(Flag, X);
+  // sub has two results while X only have one. DAG combine assumes the value
+  // type matches.
+  if (N->getNumValues() > 1)
```

Comment:
Doesn't `N` always have 2 results?

Reply:
`N` can be a `X86ISD::CMP` or `X86ISD::SUB`. When `N` is `CMP`, it only has one result.

Reply:
How about `if (N->getOpcode() == X86ISD::SUB)` for clarity?

Reply:
Good suggestion, done